### PR TITLE
:book: Builder: Improve godocs

### PR DESCRIPTION
**Pull request by alvaroaleman**
This updates the godocs in the builder to reference the current implementations of `source.Kind` correctly in the godocs that explain `For` and `Owns`.

Also removes the warning in `WatchesRawSource`. With the introduction of the typed handler and predicates, this became a valid use case.

### CODE REVIEW

File: pkg/builder/controller.go [hunk 1]
```diff
@@ -90,8 +87,9 @@ type ForInput struct {
 
 // For defines the type of Object being *reconciled*, and configures the ControllerManagedBy to respond to create / delete /
 // update events by *reconciling the object*.
+//
 // This is the equivalent of calling
-// Watches(&source.Kind{Type: apiType}, &handler.EnqueueRequestForObject{}).
+// Watches(source.Kind(cache, &Type{}, &handler.EnqueueRequestForObject{}).
```
Comment (sbueringer) [inline pkg/builder/controller.go:92]:
```suggestion
// Watches(source.Kind(cache, &Type{}, &handler.EnqueueRequestForObject{})).
```
if I'm counting correctly :)

**Review by sbueringer — COMMENTED** [review]
One nit

File: pkg/builder/controller.go (hunk 1 above)
Comment (alvaroaleman) [reply pkg/builder/controller.go]: fixed

**Comment by sbueringer** [comment]
Thank you!

/lgtm
/approve

**Pull request merged**